### ISSUE TITLE
Operator Compare crash when Mode too large

### ISSUE THREAD

**Issue by hollyhook** [still-scene/t3#45]
Connect the Compare operator input Mode with an int > 4 (?) causes a crash

```
System.ArgumentOutOfRangeException
  HResult=0x80131502
  Message=Specified argument was out of the range of valid values.
  Source=Operators
  StackTrace:
   at T3.Operators.Types.Id_026869ee_b62f_481e_aadf_f8a1db77fe65.Compare.Update(EvaluationContext context) in C:\dev\repos\t3\Operators\Types\Compare.cs:line 37
   at T3.Core.Operator.Slots.Slot' 1.Update(EvaluationContext context) in C:\dev\repos\t3\Core\Operator\Slots\Slot.cs:line 73
   at T3.Gui.OutputUi.OutputUi' 1.Recompute(ISlot slot, EvaluationContext context) in C:\dev\repos\t3\t3\Gui\OutputUi\OutputUi.cs:line 45
   at T3.Gui.OutputUi.OutputUi' 1.DrawValue(ISlot slot, EvaluationContext context, Boolean recompute) in C:\dev\repos\t3\t3\Gui\OutputUi\OutputUi.cs:line 31
   at T3.Gui.Windows.Output.OutputWindow.DrawOutput(Instance instanceForOutput, Instance instanceForEvaluation) in C:\dev\repos\t3\t3\Gui\Windows\Output\OutputWindow.cs:line 191
   at T3.Gui.Windows.Output.OutputWindow.DrawContent() in C:\dev\repos\t3\t3\Gui\Windows\Output\OutputWindow.cs:line 73
   at T3.Gui.Windows.Window.DrawOneInstance() in C:\dev\repos\t3\t3\Gui\Windows\Window.cs:line 93
   at T3.Gui.Windows.Output.OutputWindow.DrawAllInstances() in C:\dev\repos\t3\t3\Gui\Windows\Output\OutputWindow.cs:line 40
   at T3.Gui.Windows.Window.Draw() in C:\dev\repos\t3\t3\Gui\Windows\Window.cs:line 49
   at T3.Gui.Windows.WindowManager.Draw() in C:\dev\repos\t3\t3\Gui\Windows\WindowManager.cs:line 94
   at T3.Gui.T3Ui.Draw() in C:\dev\repos\t3\t3\Gui\T3UI.cs:line 92
   at T3.Program.<>c__DisplayClass9_0.<Main>g__RenderCallback|1() in C:\dev\repos\t3\t3\Program.cs:line 182
   at SharpDX.Windows.RenderLoop.Run(Control form, RenderCallback renderCallback, Boolean useApplicationDoEvents)
   at T3.Program.Main() in C:\dev\repos\t3\t3\Program.cs:line 198

```

One can argue if that is really a bug, since the value is indeed out of range, but it is not a gentle way to react (data loss etc.)

**Comment by pixtur** [comment]
Yes it is a bug an needs to be fixed.

Ideally all casts from int inputs to enums should be clamped:

```
            switch ((Modes)Mode.GetValue(context))
```

...should be...
```
            switch ((Modes)Mode.GetValue(context).Clamp(0, 2))
```

**Comment by pixtur** [comment]
Available in v3.3
